garrysmod/lua/vgui/dtextentry.lua

```diff
@@ -47,6 +48,23 @@ function PANEL:Init()
 	self.History = {}
 	self.HistoryPos = 0
 	
+	self.PlaceholderText = vgui.Create( "DLabel", self )
+	self.PlaceholderText:Dock( FILL )
+	self.PlaceholderText:DockMargin( 3, 1, 3, 0 )
+	self.PlaceholderText:SetMouseInputEnabled( false )
+	self.PlaceholderText:SetText( "" )
+	self.PlaceholderText:SetWrap( true )
+	self.PlaceholderText:SetTextColor( Color( 169, 169, 169 ) )

```

Comment:
This should be an configurable. Preferably through both Derma skins (see `SKIN:SchemeTextEntry`) and as a PANEL AccessorFunc.